src/components/signin/sagas.js

```diff
@@ -21,13 +21,19 @@ function* signinUser(action){
             },
             method: 'POST',
         });
-        if(response.ok){
+
+        if (response.ok) {
             const data = yield response.json();
-            yield effects.put(signinSuccess(data));
+            yield effects.put(signinSuccess(true));
             yield effects.put(setCookie(data));
+
+        } else {
+            const data = yield response.json();
+            yield effects.put(signinSuccess(false, data));
```

Comment:
if you are going to pass in data into `signinSuccess`, there is no need to pass in the boolean as the first parameter. Just use the information from data to determine whether the request was successful or not

Reply:
I need the two parameters because i only want a payload  in my reducer if `signinSuccess`  is false.
if i not, on successful signin, my message for invalid password, etc will be undefined thereby breaking the code

Reply:
> if you are going to pass in data into `signinSuccess`, there is no need to pass in the boolean as the first parameter. Just use the information from data to determine whether the request was successful or not

I need the two parameters because i only want a payload  in my reducer if `signinSuccess`  is false.
if i not, on successful signin, my message for invalid password, etc will be undefined thereby breaking the code

Reply:
What i am saying is that exact pattern is not correct. Your actions should be expecting the same sorts of data each time it is called. Otherwise, use a new action to do what you want.